### PR TITLE
Product search samples

### PR DESCRIPTION
**Pull request by nirupa-kumar**
## Product Search (v1p3beta1)
This Pull Request adds the code samples for cloud.google.com documentation for Product Search How-to Guides

Product Search How-to Guides –> [vision/product-search/docs/how-to](https://cloud.google.com/vision/product-search/docs/how-to)

### products.js
- [x] create_product
- [x]  get_product
- [x]  list_products
- [x]  delete_product
- [x]  update_product_labels
### productSets.js
- [x]  create_product_set
- [x]  get_product_set
- [x]  list_product_sets
- [x]  delete_product_set
#### –> Products in productSets.js
- [x]  add_product_to_product_set
- [x]  list_products_in_product_set
- [x]  remove_product_from_product_set
### referenceImages.js
- [x]  create_reference_image
- [x]  get_reference_image
- [x]  list_reference_images
- [x]  delete_reference_image
### similarProducts.js
- [x]  get_similar_products_file
- [x]  get_similar_products_uri
### importProductSets.js
- [x] import_product_images
This last sample has been newly added after looking into the python and Java samples for comparison. 

For the canonical versions, please see the [Python Product Search Samples](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/vision/cloud-client/product_search) in [python-docs-samples](https://github.com/GoogleCloudPlatform/python-docs-samples/)

These samples are in Beta and require using the v1p3beta1 API endpoint:
```javascript
const vision = require('@google-cloud/vision').v1p3beta1;
const client = new vision.ProductSearchClient();
```
Before being ready for review & merge:

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary) - working on this one - README is not getting updated with the Product Search code - looking into it.

### CODE REVIEW

**Comment by nirupa-kumar** [comment]
@beccca @ace-n @jmdobry 
Please review. 
I'm guessing since I do not have write access on the repo I am unable to add to the original pull request (https://github.com/googleapis/nodejs-vision/pull/136) Please let me know if there is anything that can be done. 
I have also added one more sample for importing product sets that was previously not included in the list. This matches up with the samples found in Java and python.
The README file generation is currently not working and I am looking into it.

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,18 +59,7 @@ jobs:
       - image: 'node:6'
         user: node
     steps: &unit_tests_steps
-      - checkout
```
Comment (beccca) [inline .circleci/config.yml]: @nirupa-kumar Do you know why these changes are showing up? There are changes to .gitignore, the CircleCI config, common.cfg files, etc etc – all seemingly unrelated to the PR

Is it maybe just out-of-date and needs to have `master` merged into this branch?

I'd like to see these items as not part of this PR (which a merge/rebase might fix?)

Comment (nirupa-kumar) [reply .circleci/config.yml]: definitely were not added as part of product search - will do the merge/rebase fix.

**Comment by beccca** [comment]
The PR currently shows **58 Commits** & **79 Files Changed**, only a few of which are related to Product Search samples

I'm not sure what caused this?

The Product Search content itself LGTM as far as I can tell, but the diff is disconcerting 😕

**Comment by nirupa-kumar** [comment]
@becca @JustinBeckwith 
Please let me know if you know of a better way to approach this. 
In short - there are three people who have worked on this, me included.The branch has not been merged into master. I have updated the branch itself with the master, but when doing a push - the 50+ commits got tagged along. One thing I am trying to do right now is doing a merge with the master and a rebase. But when doing a push (after the rebase), it gets rejected. The solutions I have found online range from "doing a force push" to "deleting the branch in remote and pushing it again". Trying to figure out which one makes more sense. 

P.S.: I do not have write access to the repo.

**Comment by beccca** [comment]
Oh! I didn't notice that this PR isn't being sent to `master`! Let me see if I can change that post-hoc...

**Comment by beccca** [comment]
THERE WE GO! Good catch @JustinBeckwith!

**Comment by beccca** [comment]
I'll LGTM pending test pass!

File: samples/productSearch/productSearch.v1p3beta1.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  // const location = 'A compute region name';
+  // const productSetId = 'Id of the product set';
+
+  var productSetPath = client.productSetPath(projectId, location, productSetId);
+
+  var request = {
+    name: productSetPath,
+  };
+
+  //client
```
Comment (beccca) [inline samples/productSearch/productSearch.v1p3beta1.js]: Remove commented out code

File: samples/productSearch/similarProducts.v1p3beta1.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        console.log('Product display name:', result['product'].displayName);
+        console.log('Product description:', result['product'].description);
+        console.log('Product category:', result['product'].productCategory);
+      });
+    })
+    .catch(err => {
+      console.error(err);
+    });
+}
+function getSimilarProductsGcs(
```
Comment (beccca) [inline samples/productSearch/similarProducts.v1p3beta1.js]: Missing `START`|`END` region tags

File: samples/productSearch/similarProducts.v1p3beta1.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function getSimilarProductsFile(
```
Comment (beccca) [inline samples/productSearch/similarProducts.v1p3beta1.js]: Missing `START`|`END` region tags

File: samples/productSearch/productSearch.v1p3beta1.js (hunk 2 above)
Comment (nirupa-kumar) [reply samples/productSearch/productSearch.v1p3beta1.js]: This has been resolved as part of https://github.com/googleapis/nodejs-vision/pull/196#

File: samples/productSearch/similarProducts.v1p3beta1.js (hunk 3 above)
Comment (nirupa-kumar) [reply samples/productSearch/similarProducts.v1p3beta1.js]: This has been resolved as part of https://github.com/googleapis/nodejs-vision/pull/196#

File: samples/productSearch/similarProducts.v1p3beta1.js (hunk 4 above)
Comment (nirupa-kumar) [reply samples/productSearch/similarProducts.v1p3beta1.js]: This has been resolved as part of https://github.com/googleapis/nodejs-vision/pull/196#

**Comment by nirupa-kumar** [comment]
Will be closing this PR , since new one with the rebased code as a direct branch on the master has been created. All review comments here have been resolved as part of https://github.com/googleapis/nodejs-vision/pull/196#
